### ISSUE TITLE
TimeDistributed BatchNormalization layer with fixed batch_shape is not working....reproducible minimal code is mentioned.

### ISSUE THREAD

**Issue by madhurgoel** [keras-team/keras#8770]
First of all Thanks guys for keeping up this awesome work.
I am using TF backend: standard PIP install. I Need the following:

RNN(return_sequences=True, stateful=True) --->> TimeDistributed( BatchNormalization() )

But for the statefulness, I need to fix the batch_size and that is where it is collapsing. following is the minimal code to reproduce the problem:

from keras.layers import *
from keras.models import *
import keras.backend as K
batch_size = 64
inputs = Input(batch_shape = (batch_size, 100, 10))
#can't rehape as it would not take batch_size into account
x = TimeDistributed(BatchNormalization())(inputs)
model = Model(inputs=inputs, outputs=x)
model.compile(loss='mae', optimizer='sgd')
X = np.random.rand(batch_size, 100, 10)
Y = np.random.rand(batch_size, 100, 10)
history = model.fit(X, Y,batch_size=batch_size, epochs=4)

OUTPUT : InvalidArgumentError (see above for traceback): You must feed a value for placeholder tensor 'time_distributed_1/keras_learning_phase' with dtype bool
	 [[Node: time_distributed_1/keras_learning_phase = Placeholder[dtype=DT_BOOL, shape=<unknown>, _device="/job:localhost/replica:0/task:0/cpu:0"]()]]

Any help would be really appreciable, Thanks.

**Comment by madhurgoel** [comment]
setting batch_size to None i.e inputs = Input(batch_shape = (None, 100, 10)) is working fine. But for statefulness of RNN I would require to fix batch_size.

**Comment by guofuzheng** [comment]
@madhurgoel Hello , have you fixed the problem ? My network is similar to yours , i need the output of TimeDistributed to compute parameters . And the shape of input1 is (10,128,128,3) , the input in model.fit() is (644,10,128,128,3) . I met the same problem , may i ask for your advice?

**Comment by madhurgoel** [comment]
Sorry to say, but I was not able to overcome this issue. It forced me to design the whole this in raw TensorFlow.

**Comment by guofuzheng** [comment]
Okay , thank you for your help .
